crates/subspace-farmer/src/single_disk_plot/piece_receiver.rs

```diff
@@ -0,0 +1,236 @@
+use crate::RpcClient;
+use async_trait::async_trait;
+use std::collections::HashSet;
+use std::error::Error;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::time::Duration;
+use subspace_core_primitives::{Piece, PieceIndex, PieceIndexHash};
+use subspace_networking::libp2p::PeerId;
+use subspace_networking::utils::multihash::MultihashCode;
+use subspace_networking::{Node, PieceByHashRequest, PieceKey, ToMultihash};
+use tokio::time::sleep;
+use tracing::{debug, error, info, trace};
+
+/// Defines a duration between get_piece calls.
+const GET_PIECE_WAITING_DURATION_IN_SECS: u64 = 1;
+
+#[async_trait]
+pub trait PieceReceiver {
+    async fn get_piece(
+        &mut self,
+        piece_index: PieceIndex,
+    ) -> Result<Option<Piece>, Box<dyn Error + Send + Sync + 'static>>;
+}
+
+#[async_trait]
+impl PieceReceiver for BenchPieceReceiver {
+    async fn get_piece(
+        &mut self,
+        _piece_index: PieceIndex,
+    ) -> Result<Option<Piece>, Box<dyn Error + Send + Sync + 'static>> {
+        Ok(Some(self.piece.clone()))
+    }
+}
+
+pub struct BenchPieceReceiver {
+    piece: Piece,
+}
```

Comment:
Let's not have impls before corresponding definition and please move this into benches, we ideally shouldn't expose bench- and test-specific stuff publicly if possible and in this case it is certainly possible.